internal/provider/providers/duckdns/provider.go

```diff
@@ -146,7 +146,11 @@ func (p *Provider) Update(ctx context.Context, client *http.Client, ip netip.Add
 	case s[0:minChars] == "KO":
 		return netip.Addr{}, fmt.Errorf("%w", errors.ErrAuth)
 	case s[0:minChars] == "OK":
-		ips := ipextract.IPv4(s)
+		if ip.Is6() {
+			ips := ipextract.IPv6(s)
```

Comment:
There seems to be an issue with variable scope here.
As I do not know the language go, I would appreciate help of an expert.

Reply:
Using `:=` declares a new variable, which would only be local within the if block scope, so we need to declare it earlier and then re-assign it with `=` (as you've seen from the no ip code).